### ISSUE TITLE
Terraform crashes when tag-protection is manually deleted between applies

### ISSUE THREAD

**Issue by phgogo** [gitlabhq/terraform-provider-gitlab#779]
Hi there,

### Terraform Version
```
Terraform v1.0.9
on linux_amd64
+ provider registry.terraform.io/gitlabhq/gitlab v3.8.0
```

### Affected Resource(s)

- gitlab_tag_protection

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a `gitlab_tag_protection` resource
2. Manually delete the tag protection in GitLab, until there are no tag-protections left in the project
3. Try to Apply again

### Expected Behavior
The tag protection should be re-created

### Actual Behavior
Terraform crashes, since it can not refresh the protection, since the /projects endpoint does not offer the sub-page `/protected_tags`

```

│ Error: GET https://my-gitlab-instance.com/api/v4/projects/333/protected_tags/*: 404 {message: 404 Not found}
│ 
│   with module.gitlab_project-library_projects["gitlab-ci-templates.general.pipeline-include-library"].gitlab_tag_protection.default,
│   on .terraform/modules/gitlab_project-library_projects/main.tf line 72, in resource "gitlab_tag_protection" "default":
│   72: resource "gitlab_tag_protection" "default" {
│ 

```

**Comment by slimm609** [comment]
This also happens with webhooks

**Comment by armsnyder** [comment]
This will likely be fixed in the next release by #318

**Comment by timofurrer** [comment]
#318 has been merged. I suggest that we keep our fingers crossed and if the issue pops up again after 3.9.0 we can open another issue 🍰
